router/batchrouter/asyncdestinationmanager/asyncdestinationmanager.go

```diff
@@ -266,30 +124,12 @@ func GetMarshalledData(payload string, jobID int64) string {
 	return string(responsePayload)
 }
 
-func GenerateFailedPayload(config map[string]interface{}, jobs []*jobsdb.JobT, importID, destType, csvHeaders string) []byte {
-	var failedPayloadT AsyncFailedPayload
-	failedPayloadT.Input = make([]map[string]interface{}, len(jobs))
-	index := 0
-	failedPayloadT.Config = config
-	for _, job := range jobs {
-		failedPayloadT.Input[index] = make(map[string]interface{})
-		var message map[string]interface{}
-		metadata := make(map[string]interface{})
-		err := json.Unmarshal([]byte(GetTransformedData(job.EventPayload)), &message)
-		if err != nil {
-			panic("Unmarshalling Transformer Data to JSON Failed")
-		}
-		metadata["job_id"] = job.JobID
-		failedPayloadT.Input[index]["message"] = message
-		failedPayloadT.Input[index]["metadata"] = metadata
-		index++
-	}
-	failedPayloadT.DestType = strings.ToLower(destType)
-	failedPayloadT.ImportId = importID
-	failedPayloadT.MetaData = MetaDataT{CSVHeaders: csvHeaders}
-	payload, err := json.Marshal(failedPayloadT)
-	if err != nil {
-		panic("JSON Marshal Failed" + err.Error())
+func NewManager(destination *backendconfig.DestinationT, backendConfig backendconfig.BackendConfig) Asyncdestinationmanager {
+	destType := destination.DestinationDefinition.Name
+	if destType == "BINGADS_AUDIENCE" {
+		return bingads.NewManager(destination, backendConfig, HTTPTimeout)
+	} else if destType == "MARKETO_BULK_UPLOAD" {
+		return marketobulkupload.NewManager(destination, HTTPTimeout)
```

Comment:
[minor] Personal preference for readability:
```suggestion
	switch destination.DestinationDefinition.Name {
    case "BINGADS_AUDIENCE":
		return bingads.NewManager(destination, backendConfig, HTTPTimeout)
	case "MARKETO_BULK_UPLOAD": 
		return marketobulkupload.NewManager(destination, HTTPTimeout)
```